dtfs-central-api/src/services/repositories/utilisation-reports-repo.test.js

```diff
@@ -37,6 +96,34 @@ describe('utilisation-reports-repo', () => {
         id: '123',
         name: 'test user',
       },
+    }, {
+      bank: {
+        id: '124',
+        name: 'test bank',
+      },
+      month: 4,
+      year: 2021,
+      dateUploaded: expect.any(Date),
+      path: 'test path',
+      uploadedBy: {
+        id: '123',
+        name: 'test user',
+      },
+    }];
+
+    it('filters and sorts the data correctly', async () => {
+      const toArraySpy = jest.fn(() => (mockUtilisationReports));
+      const findSpy = jest.fn(() => ({
+        toArray: toArraySpy,
+      }));
+      const getCollectionMock = jest.fn(() => ({
+        find: findSpy,
+      }));
+      jest.spyOn(db, 'getCollection').mockImplementation(getCollectionMock);
+
+      const utilisationReports = await getUtilisationReportDetails(bankId);
+      expect(utilisationReports[0]).toEqual(mockUtilisationReports[3]);
+      expect(utilisationReports[utilisationReports.length - 1]).toEqual(mockUtilisationReports[0]);
```

Comment:
It's not immediately clear what we're asserting on here, and I think we could go further to assert that the data is correctly ordered by year then month.

We could instead do something like this:
```
const testUtilisationReport = {
  bank: {
    id: '124',
    name: 'test bank',
  },
  month: 4,
  year: 2021,
  dateUploaded: expect.any(Date),
  path: 'test path',
  uploadedBy: {
    id: '123',
    name: 'test user',
  },
};

it('sorts the data by year then month', async () => {
  // Arrange
  const bankId = testUtilisationReport.bank.id;
  const report1 = { ...testUtilisationReport, month: 2, year: 2022 };
  const report2 = { ...testUtilisationReport, month: 3, year: 2021 };
  const report3 = { ...testUtilisationReport, month: 1, year: 2022 };
  const report4 = { ...testUtilisationReport, month: 2, year: 2021 };

  const mockUtilisationReports = [report1, report2, report3, report4];

  spyOn(db, 'getCollection').mockImplementation(() => ({
    find: jest.fn(() => ({
      toArray: jest.fn(() => mockUtilisationReports),
    })),
  }));

  // Act
  const response = await getUtilisationReportDetails(bankId);

  // Assert
  const expectedResponse = [report4, report2, report3, report1];
  expect(response).toEqual(expectedResponse);
});
```
The difference here is that when I create the mock reports in the test I'm only setting the values that I care about for the test (`month` and `year` in this case) so it's clear what we're testing. I can then use these mock reports to more easily check the entire response.

Reply:
Ah yes I like this much more, I wanted something which made it more obvious but didn't know how to do it, thanks 🙏